### PR TITLE
Upgrades ginkgo from v1.16.5 to v2.1.3

### PR DESCRIPTION
**Pull request by spring1843**
**1. Issue, if available:**
Ginkgo 1.* are deprecated, and v2 offers some nice features:
https://github.com/onsi/ginkgo/blob/master/docs/MIGRATING_TO_V2.md#major-changes

**2. Description of changes:**
Upgrades ginkgo from `v1.16.5` to `v2.1.3`

Follows steps described in:
https://github.com/onsi/ginkgo/blob/master/docs/MIGRATING_TO_V2.md

**3. How was this change tested?**

**4. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [X] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by tzneal** [comment]
There should be some changes to the Makefile to go along with this, several of the flags are renamed and the old names are deprecated, try `make battletest`.

File: Makefile [hunk 1]
```diff
@@ -31,8 +31,8 @@ battletest: ## Run stronger tests
 	gocyclo -over 11 ./pkg
 	# Run randomized, parallelized, racing, code coveraged, tests
 	ginkgo -r \
-		-cover -coverprofile=coverage.out -outputdir=. -coverpkg=./pkg/... \
-		--randomizeAllSpecs --randomizeSuites -race
+		--cover --coverprofile=coverage.out --output-dir=.\
+		--randomize-all --randomize-suites --race --trace
```
Comment (tzneal) [inline Makefile:35]: Try adding `-coverpkg=github.com/aws/karpenter/pkg/...` , not sure why the -coverpkg=./pkg/... isn't working correctly.

File: Makefile [hunk 2]
```diff
@@ -31,8 +31,8 @@ battletest: ## Run stronger tests
 	gocyclo -over 11 ./pkg
 	# Run randomized, parallelized, racing, code coveraged, tests
 	ginkgo -r \
-		--cover --coverprofile=coverage.out --output-dir=.\
-		--randomize-all --randomize-suites --race --trace
+		-cover -coverprofile=coverage.out \
+		-race  --randomize-all --randomize-suites --trace
```
Comment (ellistarn) [inline Makefile:35]: nit: is there any extra space after --race?

**Comment by spring1843** [comment]
I'm seeing some odd behavior in coverage data, it doesn't look like the new version is doing a stable job on calculating the coverage, I'll reopen this once those problems are fixed.

**Pull request closed without merging**